pkg/policy/execute.go

```diff
@@ -119,65 +118,56 @@ func (e *Executor) with(policy string, args ...interface{}) *Executor {
 		return nil, err
 	}
-	var rest []string
-	if len(selector) > 0 {
-		rest = selector[1:]
-	}
-	var found bool
 	total := ExecutionResult{PolicyName: req.Policy.Name, Passed: true, Results: make([]*QueryResult, 0)}
 	for _, p := range policy.Policies {
-		if len(selector) == 0 || p.Name == selector[0] {
-			found = true
-			executor := e.with(p.Name)
-			executor.log.Info("starting policy execution")
-			r, err := executor.Execute(ctx, req, p, rest)
-			if err != nil {
-				executor.log.Error("failed to execute policy", "err", err)
-				return nil, fmt.Errorf("%s/%w", policy.Name, err)
-			}
-			total.Passed = total.Passed && r.Passed
-			total.Results = append(total.Results, r.Results...)
-			if !total.Passed && req.StopOnFailure {
-				return &total, nil
-			}
-
+		executor := e.with(p.Name)
+		executor.log.Info("starting policy execution")
+		r, err := executor.Execute(ctx, req, p)
+		if err != nil {
+			executor.log.Error("failed to execute policy", "err", err)
+			return nil, fmt.Errorf("%s/%w", policy.Name, err)
+		}
+		total.Passed = total.Passed && r.Passed
+		total.Results = append(total.Results, r.Results...)
+		if !total.Passed && req.StopOnFailure {
+			return &total, nil
 		}
 	}
 
 	for _, q := range policy.Checks {
-		if len(selector) == 0 || q.Name == selector[0] {
-			found = true
-			e.log = e.log.With("query", q.Name)
-			qr, err := e.executeQuery(ctx, q)
-			if err != nil {
-				e.log.Error("failed to execute query", "err", err)
-				return nil, fmt.Errorf("%s/%w", policy.Name, err)
-			}
-			total.Passed = total.Passed && qr.Passed
-			total.Results = append(total.Results, qr)
-			e.log.Info("Check finished with result", "passed", qr.Passed)
-			if e.progressUpdate != nil {
-				e.progressUpdate(Update{
-					FinishedQueries: 1,
-				})
-			}
-			if !total.Passed && req.StopOnFailure {
-				return &total, nil
-			}
+		e.log = e.log.With("query", q.Name)
+		qr, err := e.executeQuery(ctx, q)
+		if err != nil {
+			e.log.Error("failed to execute query", "err", err)
+			return nil, fmt.Errorf("%s/%w", policy.Name, err)
+		}
+		total.Passed = total.Passed && qr.Passed
+		total.Results = append(total.Results, qr)
+		e.log.Info("Check finished with result", "passed", qr.Passed)
+		if e.progressUpdate != nil {
+			e.progressUpdate(Update{
+				FinishedQueries: 1,
+			})
+		}
+		if !total.Passed && req.StopOnFailure {
+			return &total, nil
 		}
 	}
-	if !found && len(selector) > 0 {
-		e.log.Error("policy/query not found with provided sub-policy selector", "selector", selector, "available_policies", policy.Policies.All())
-		return nil, fmt.Errorf("%s//%s: %w", policy.Name, path.Join(selector...), ErrPolicyOrQueryNotFound)
-	}
+	// if !found && len(selector) > 0 {
+	// 	e.log.Error("policy/query not found with provided sub-policy selector", "selector", selector, "available_policies", policy.Policies.All())
+	// 	return nil, fmt.Errorf("%s//%s: %w", policy.Name, path.Join(selector...), ErrPolicyOrQueryNotFound)
+	// }
```

Comment:
if this is commented out either remove it or uncomment it, or add a todo explaining why its commented out

Reply:
removed